payment_methods/SquareOnsite/forms/SettingsForm.php

```diff
@@ -184,42 +126,46 @@ public function validateConnection(EE_Payment_Method $pmInstance)
         $squareData = $pmInstance->get_extra_meta(Domain::META_KEY_SQUARE_DATA, true);
         $pmDebugMode = $pmInstance->debug_mode();
         $debugInput = $this->get_input('PMD_debug_mode', false);
-        $modeInput = $this->get_input(Domain::META_KEY_AUTH_TYPE, false);
         if (isset($squareData[ Domain::META_KEY_USING_OAUTH ]) && $squareData[ Domain::META_KEY_USING_OAUTH ]) {
-            if (isset($squareData[ Domain::META_KEY_LIVE_MODE ])
+            // First check the credentials and the API connection
+            $oauthHealthCheck = $this->oauthHealthCheck($pmInstance);
+            // and reset the OAuth connection in case we are no longer authorized for some reason.
+            if (isset($oauthHealthCheck['error'])) {
+                if ($oauthHealthCheck['error']['code'] === 'NOT_AUTHORIZED') {
+                    // Seems like the Token got revoked or outdated, reset the connection.
+                    $oauthReset = $this->resetOauthSettings($pmInstance);
+                    if ($oauthReset) {
+                        $this->doValidationError(
+                            'The OAuth %1$sauthorization was revoked%2$s so the connection was reset. Please re-authorize (Connect) for the Square payment method to function properly.',
+                            'eea_square_oauth_connection_was_reset'
+                        );
```

Comment:
translations need to operate on the actual strings else they can't be parsed 

```suggestion
                        $this->doValidationError(
                        	 esc_html__(
                        	 	'The OAuth %1$sauthorization was revoked%2$s so the connection was reset. Please re-authorize (Connect) for the Square payment method to function properly.',
                        	 	'event_espresso'
                        	 ),
                            'eea_square_oauth_connection_was_reset'
                        );
```